### DISCUSSION TITLE
how to get indefinite fields in json request body？

### DISCUSSION THREAD

**Discussion by fenghaoyu** [fastapi/fastapi#7912] [Questions]
client send json params like this:
`{
    "id":1,
    "a":"something"
}`
the “a” is indefinite. maybe it's "a" mabey it's "b".
so, how can i get those indefinite fields？

**Comment by Dustyposa** [comment]
Change your request body, like below:
```
{ "id":1, "data": ["a":"something"]}
```
Or directly use [`starlette.requests`](https://www.starlette.io/requests/)
:smile:

**Comment by fenghaoyu** [comment]
thanks very very much.


发自我的iPhone

------------------ Original ------------------

**Comment by fenghaoyu** [comment]
i'm doing this to solve it:
```   
class Item(BaseModel):
    def __init__(self, **data):
        super().__init__(**data)

        extra_raw_data_keys = data.keys() - self.__fields__.keys()
        object.__setattr__(self, 'extra_raw_data', {k: data.get(k) for k in extra_raw_data_keys})

@router.post('/')
async def apply_stage(item: Item):
    print(item.extra_raw_data)
```

**Comment by tiangolo** [comment]
Thanks for the help here @Dustyposa ! :bowing_man: :cake: 

And thanks @fenghaoyu for reporting back and closing the issue :heavy_check_mark:
